pkg/provider/gcp/secretmanager/secretsmanager.go

```diff
@@ -153,12 +154,21 @@ func (sm *ProviderGCP) GetSecret(ctx context.Context, ref esv1alpha1.ExternalSec
 		return nil, fmt.Errorf(errClientGetSecretAccess, err)
 	}
 
-	err = sm.SecretManagerClient.Close()
-	if err != nil {
-		return nil, fmt.Errorf(errClientClose, err)
```

Comment:
Had to remove the closing of the connection, because when we iterate over the keyrefs the previous connection was still being closed when trying to get the next sercret. We probably need to open an issue to change our interface to include a wait method or something at the beginning of the GetSecret method.

Reply:
Can we defer it instead?

Reply:
ah it needs to be outside of the `GetSecret()` function.

Reply:
I changed the generic provider to have a Close method. So for clients that need closing we can call it on the controller: https://github.com/external-secrets/external-secrets/pull/217/commits/01267ea6aa898544547575d2466727ee9f6304a9

Seems to make sense to me. What do you think?

Reply:
Seems to work :) https://github.com/external-secrets/external-secrets/runs/2914847394?check_suite_focus=true